src/main/java/cz/student/adambrg/UserDao.java

```diff
@@ -0,0 +1,139 @@
+package cz.student.adambrg;
+
+import cz.common.DbUtil;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    public static final String CREATE_USER =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ? )";
+
+
+    public User create(User user) { //metoda pro vytvareni uzivatele
+        try (Connection conn = DbUtil.getConnection()) {
+            PreparedStatement statement = conn.prepareStatement(CREATE_USER, PreparedStatement.RETURN_GENERATED_KEYS);
+
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getInt(1));
+                System.out.println("User ID " + user.getId());
+
+            }
+            return user;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while creating a user");
```

Comment:
CLEAN-CODE: dobry jen bych otocil, prvni bych zalogoval obecne info a pak stacktrace. Opravit u vsech vyskytu.

```suggestion
            System.err.println("ERROR while creating a user");
            e.printStackTrace();
```

Reply:
opraveno